streaming/base/util.py

```diff
@@ -358,6 +358,43 @@ def _merge_index_from_list(index_file_urls: List[Union[str, Tuple[str, str]]],
         if not keep_local:
             shutil.rmtree(cu.local, ignore_errors=True)
 
+def _not_merged_index(index_file_path: str, out: str):
+    """Check if index_file_path is the merged index at folder out.
+
+    Args:
+        index_file_path (str): the path to index.json file
+        out (str): remote or local url of a folder
+    Return:
+        (bool): no if index.json sits in out instead of in the subfolders of out
+    """
+    prefix = str(urllib.parse.urlparse(out).path)
+    return os.path.dirname(index_file_path).strip('/') != prefix.strip('/')
+
+
+def _format_remote_index_files(remote: str, files: List[str]) -> List[str]:
+    """Formats the remote index files by appending the remote URL scheme and netloc to each file.
+
+    Args:
+        remote (str): The remote URL.
+        files (list[str]): The list of files.
+
+    Returns:
+        list[str]: The formatted remote index files.
+    """
+    remote_index_files = []
+    obj = urllib.parse.urlparse(remote)
+    for file in files:
+        if file.endswith(get_index_basename()) and _not_merged_index(file, remote):
+            join_char = '://'
+            if obj.scheme == 'dbfs':
+                path = Path(remote)
+                prefix = os.path.join(path.parts[0], path.parts[1])
+                if prefix == 'dbfs:/Volumes':
+                    join_char = '/'
```

Comment:
I think here it also should be :/. Can you change and add a "dbfs" scheme to the parametrizer list of your unit test?

Reply:
Indeed, I added the test and fixed the bug